src/NUnitFramework/nunit.framework.legacy/LegacyExtensions.Types.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using System;
+
+// ReSharper disable once CheckNamespace
+namespace NUnit.Framework
+{
+    /// <summary>
+    /// C#14 static extension methods for type assertions
+    /// </summary>
+    public static class LegacyAssertTypeExtensions
+    {
+        extension(Assert)
+        {
+            #region IsAssignableFrom
+
+            public static void IsAssignableFrom(Type expected, object? actual, string? message = null, params object?[]? args)
+                => Legacy.ClassicAssert.IsAssignableFrom(expected, actual, message ?? string.Empty, args);
+
+            public static void IsAssignableFrom(Type expected, object? actual)
+                => Legacy.ClassicAssert.IsAssignableFrom(expected, actual);
```

Comment:
The _message_ should not be optional, you already have the overload without a _message_ parameter.